contracts/cosmwasm-vm/cw-xcall/src/execute_call.rs

```diff
@@ -87,13 +86,17 @@ impl<'a> CwCallService<'a> {
                     CallServiceResponseType::CallServiceResponseSuccess,
                 );
                 let event = event_call_executed(req_id, code, "success");
+                self.remove_proxy_request(deps.storage, req_id);
                 (message_response, event)
             }
             cosmwasm_std::SubMsgResult::Err(err) => {
                 let code = CallServiceResponseType::CallServiceResponseFailure;
                 let error_message = format!("CallService Reverted : {err}");
                 let message_response = CSMessageResult::new(request.sequence_no(), code.clone());
                 let event = event_call_executed(req_id, code.into(), &error_message);
```

Comment:
I think it would be better to just revert the transaction instead, so send the execute with reply on success only mabye?

Reply:
this is the reply from dapp so we have to handle both cases to send response and cleanup if its not a persisted message.